src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -132,10 +146,25 @@ private static TypeSymbol InferReturnType(BoundBlock block, Binder binder, bool
 
             // Async:
 
+            NamedTypeSymbol taskType = null;
+            var delegateReturnType = delegateType?.GetDelegateType()?.DelegateInvokeMethod?.ReturnType as NamedTypeSymbol;
+            if ((object)delegateReturnType != null)
+            {
+                MethodSymbol createBuilderMethod;
+                var delegateTaskBuilderType = delegateReturnType.GetAsyncMethodBuilderType(out createBuilderMethod);
```

Comment:
Might delegateReturnType be null in case of a void-returning delegate, or error, or other reason?